source/NVDAObjects/UIA/edge.py

```diff
@@ -409,6 +448,26 @@ def _get_isCurrent(self):
 			return valueOfAriaCurrent
 		return False
 
+	def _get_placeholder(self):
+		ariaProperties=splitUIAElementAttribs(self.UIAElement.currentAriaProperties)
+		ariaPlaceholder = ariaProperties.get('placeholder', None)
+		return ariaPlaceholder
+
+	def _get__isTextEmpty(self):
+		ti = self.makeTextInfo(textInfos.POSITION_FIRST)
+		ti.move(textInfos.UNIT_CHARACTER, 1, endPoint="end")
+		# we were unable to expand the range, it is collapsed
+		if ti.isCollapsed:
+			return True
+		# NOTE: we can not check the result of the EdgeTextInfo move implementation to determine if we added
+		# any characters to the range, since it seems to return 1 even when the text property has not changed.
+		ti.move(textInfos.UNIT_CHARACTER, 1, endPoint="end")
+		text = ti.text
+		# Edge can report a range of values for empty fields:
+		if text == " " or text == "\n" or text =="\n " or text=="\n\n" or text ==" \n":

```

Comment:
This actually required a bit of rework, I was curious about the performance optimisation that checking `len` might have, but found that it actually takes longer than just checking `text in`. Not that it matters now the function has been re-written.